p2p/src/sync/peer.rs

```diff
@@ -308,7 +314,6 @@ where
 
     async fn handle_header_list(&mut self, headers: Vec<SignedBlockHeader>) -> Result<()> {
         log::debug!("Headers list from peer {}", self.id());
-        self.last_activity = Some(self.time_getter.get_time());
```

Comment:
I changed when we set `last_activity`. It is now set only _after_ we have resolved a peer action. For example, when a peer sends us a block response, we no longer update the activity immediately, but only after we have processed the response.